src/main/java/org/medicmobile/webapp/mobile/ChtExternalApp.java

```diff
@@ -268,38 +269,56 @@ private JSONObject parseBundleToJson(Bundle bundle) {
 			return null;
 		}
 
-		private void setValueInJson(String key, JSONObject json, Bundle bundle) {
-			try {
-				Object value = bundle.get(key);
+		private void setValueInJson(String key, JSONObject json, Bundle bundle) throws IOException, JSONException {
+			Object value = bundle.get(key); // NOSONAR
 
-				if (value instanceof Bitmap) {
-					json.put(key, parseBitmapImageToBase64((Bitmap) value));
-					return;
-				}
+			if (value instanceof Bitmap bitmap) {
+				json.put(key, parseBitmapImageToBase64(bitmap, false));
+				return;
+			}
 
-				if (value instanceof Bundle) {
-					json.put(key, parseBundleToJson((Bundle) value));
-					return;
-				}
+			if (value instanceof Bundle b) {
+				json.put(key, parseBundleToJson(b));
+				return;
+			}
 
-				if (isBundleList(value)) {
-					JSONArray jsonArray = ((List<Bundle>) value)
-							.stream()
-							.map(this::parseBundleToJson)
-							.collect(Collector.of(JSONArray::new, JSONArray::put, JSONArray::put));
+			if (isBundleList(value)) {
+				JSONArray jsonArray = ((List<Bundle>) value)
+						.stream()
+						.map(this::parseBundleToJson)
+						.collect(Collector.of(JSONArray::new, JSONArray::put, JSONArray::put));
 
-					json.put(key, jsonArray);
-					return;
-				}
+				json.put(key, jsonArray);
+				return;
+			}
 
-				Optional<Uri> imagePath = getImageUri(value);
-				if (imagePath.isPresent()) {
-					json.put(key, getImageFromStoragePath(imagePath.get()));
-					return;
-				}
+			Optional<List<?>> primitiveListOpt = asPrimitiveList(value);
+			if (primitiveListOpt.isPresent()) {
+				// ODK/Enketo models a primitive multi-value list (e.g. a select_multiple question) as a
+				// space-delimited string.
+				String nodeList = primitiveListOpt
+					.stream()
+					.flatMap(List::stream)
+					.map(Object::toString)
+					.map(s -> s.replace(" ", "_"))
+					.collect(Collectors.joining(" "));
+				json.put(key, nodeList);
+				return;
```

Comment:
From my understanding of the issue you opened in [cht-core](https://github.com/medic/cht-core/issues/10508) you wanted to move serialization of data out of cht-android. 

Isn't that this part ?